src/vmm/src/devices/virtio/mmio.rs

```diff
@@ -186,10 +186,18 @@ impl MmioTransport {
             DRIVER_OK if self.device_status == (ACKNOWLEDGE | DRIVER | FEATURES_OK) => {
                 self.device_status = status;
                 let device_activated = self.locked_device().is_activated();
-                if !device_activated && self.are_queues_valid() {
-                    self.locked_device()
-                        .activate(self.mem.clone())
-                        .expect("Failed to activate device");
+                if !device_activated
+                    && self.are_queues_valid()
+                    && self.locked_device().activate(self.mem.clone()).is_err()
```

Comment:
nit: I don't quite like having a stateful call (activate) inside an `if` condition. I guess it reduces nesting, but may be a little harder to read.

Reply:
Me neither, but clippy insists :(

Reply:
It's our lucky day, the logging means clippy no longer insists on things being this way